### ISSUE TITLE
VoiceOver cannot enter into date range input with native controls

### ISSUE THREAD

**Issue by FelFly** [Esri/calcite-design-system#8460]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

While I understand Tab is the documented keyboard navigation for this component, Tab isn't as commonly used in VoiceOver as it is in say, NVDA. When trying to use the VO keys to move forward Ctrl+Option+right-arrow, I cannot find a method using native keyboard shortcuts to manually type a value nor trigger the calendar widget. (Lacking the correct keyboard to navigate via natural reading order - I cannot say whether this is applicable to JAWS.)

### Actual Behavior

While I can get focus to an input, using the VoiceOver suggestion for Ctrl+Option+Space does not result in any observable interactions and when I try Ctrl+Shift+Down-Arrow to see if I can get inside the group, it does tell me I'm in an editable text field, but still I cannot manually type in numbers in the suggested format.

![Screenshot 2023-12-19 at 5 16 22 PM](https://github.com/Esri/calcite-design-system/assets/7389593/22744e69-8506-4f8d-9f75-35abc1c2fb95)

![Screenshot 2023-12-19 at 5 24 14 PM](https://github.com/Esri/calcite-design-system/assets/7389593/03da87ac-d440-4848-aba5-8e61628a4b32)

### Expected Behavior

I would expect that I could use the VoiceOver keys to navigate into the input and then manually type in numbers if I so wished.

### Reproduction Sample

https://codepen.io/FelFly/pen/poYzNMq

### Reproduction Steps

STR
1. Using VoiceOver navigate to form input via Ctrl+Option+Right-Arrow
2. Attempt to enter into input without use of Tab or to type manual date entry in MM/DD/YYYY format with keyboard numbers

### Reproduction Version

1.11.0

### Working W3C Example/Tutorial

https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/

### Relevant Info

Safari: Version 17.1 (18616.2.9.11.9, 18616)
MacOS: Ventura 13.6.1 

### Priority impact

p4 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**Comment by geospatialem** [comment]
Spike to determine the structure of the component and how it relates to VoiceOver and next steps which we can work towards - including a time estimate.

**Comment by driskull** [comment]
I am able to get into the combobox and start typing, however it requires an additional CTR-OPT-SPACE in order to start typing.

My guess is that the way we setup our combobox has the "role" on the custom element host whereas the W3C examples have the role on the input element. We may want to consider moving the role information onto the input element.

**Comment by DitwanP** [comment]
I have not been able to successfully navigate to the input and start typing like the W3C example. 

https://github.com/Esri/calcite-design-system/assets/60022782/355dd924-f12b-42d0-98eb-675baebfc34a

In the screen recording you can see me navigate with VO to the inputs but once over them you don't see anything being typed even though I'm trying to enter numbers. In order to type I still have to do what @driskull suggested in the comment above, enter an additional CTR-OPT-SPACE.

**Comment by DitwanP** [comment]
🍭 Verified on `main`. 

I am able to:
> "use the VoiceOver keys to navigate into the input and then manually type in numbers if I so wished."

It's not exactly like the W3C example but the original issue of not being able to get into the input at all with VO keys seems to have been fixed.
